tools/sdk-versioner/src/main.rs

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use anyhow::bail;
+use std::ffi::OsStr;
+use std::fs;
+use std::path::{Path, PathBuf};
+use structopt::StructOpt;
+use toml::value::{Table, Value};
+
+const AWS_CONFIG: &str = "aws-config";
+const AWS_RUNTIME_CRATES: &[&str] = &[
+    "aws-endpoint",
+    "aws-http",
+    "aws-hyper",
+    "aws-sig-auth",
+    "aws-sigv4",
+    "aws-types",
+];
+const SDK_PREFIX: &str = "aws-sdk-";
+const SMITHY_PREFIX: &str = "aws-smithy-";
+
+#[derive(StructOpt, Debug)]
+#[structopt(
+    name = "sdk-versioner",
+    about = "CLI tool to recursively update SDK/Smithy crate references in Cargo.toml files"
+)]
+struct Opt {
+    /// Path(s) to recursively update Cargo.toml files in
+    #[structopt()]
+    crate_paths: Vec<String>,
```

Comment:
What about typing them as paths too?
```suggestion
    crate_paths: Vec<PathBuf>,
```